packages/components/atoms/f-button/src/components/Button.vue

```diff
@@ -3,44 +3,51 @@
         :is="componentType"
         :class="[
             $style['o-btn'],
-            (isIcon ? $style['o-btn--icon'] : ''),
             $style[`o-btn--${buttonType}`],
-            $style[`o-btn--size${buttonSizeClassname}`],
-            (isFullWidth ? $style['o-btn--fullWidth'] : ''),
-            (isLoading ? $style['o-btn--loading'] : '')
+            $style[`o-btn--size${buttonSizeClassname}`], {
+                [$style['o-btn--icon']]: isIcon,
+                [$style['o-btn--fullWidth']]: isFullWidth,
+                [$style['o-btn--loading']]: isLoading
+            }
         ]"
         :action-type="buttonActionType"
         :data-test-id="`${componentType}-component`"
         v-bind="$attrs"
         :aria-live="getAriaLive"
         :aria-busy="isLoading"
         v-on="!isLoading && $listeners">
-        <span
-            v-if="isLoading"
-            :class="$style['c-spinner']"
-            :data-test-id="`${componentType}-spinner`" />
-
-        <span
-            :class="[$style['o-button-content'], {
-                [$style['o-btn-content--hidden']]: isLoading
-            }]">
+        <template v-if="hasNestedContent">
+            <span
+                v-if="isLoading"
+                :class="$style['c-spinner']"
+                :data-test-id="`${componentType}-spinner`" />
 
             <span
-                v-if="hasLeadingIcon"
-                :class="[$style['o-btn-icon'], $style['o-btn-icon--leading']]"
-                data-test-id="button-leading-icon">
-                <slot name='leading-icon' />
+                :class="[$style['o-button-content'], {
+                    [$style['o-btn-content--hidden']]: isLoading
+                }]">
+
+                <span
+                    v-if="hasLeadingIcon"
+                    :class="[$style['o-btn-icon'], $style['o-btn-icon--leading']]"
+                    data-test-id="button-leading-icon">
+                    <slot name='leading-icon' />
+                </span>
+
+                <slot />
+
+                <span
+                    v-if="hasTrailingIcon"
+                    :class="[$style['o-btn-icon'], $style['o-btn-icon--trailing']]"
+                    data-test-id="button-trailing-icon">
+                    <slot name='trailing-icon' />
+                </span>
             </span>
+        </template>
 
+        <template v-else>
             <slot />
```

Comment:
I think we still want `o-button-content` class to be applied on a wrapper around the slot even if `hasNestedContent === false`

Reply:
To keep the content centered? I guess I could use the same computed property to apply the `o-button-content` class to the button itself?

For a bit of context, the issue I'm aiming to resolve here is that on the menu we need to centre content and the nested spans would mean that we have to nest 3-4 levels deep and we'd like to avoid that if possible.

Reply:
I think it _should_ be ok without the class – if the visual regression tests aren't failing then should ok ok.